tests/test_core.py

```diff
@@ -9401,6 +9401,34 @@ def test_js_library_i64_params(self):
     self.emcc_args += ['--js-library=' + test_file('core/js_library_i64_params.js')]
     self.do_core_test('js_library_i64_params.c')
 
+  def test_rust_gxx_personality_v0(self):
+    create_file('main.c', r'''
+      #include <stdio.h>
+      void __gxx_personality_v0();
+      void rust_eh_personality(){
+        __gxx_personality_v0();
+      }
+      int main(int argc, char** argv) {
+        printf("result: %d\n", argc);
+        if(argc == 2){
+          rust_eh_personality();
+        }
+        return 0;
+      }
+    ''')
+    self.emcc('main.c', ['-c'])
```

Comment:
IIRC that `__gxx_personality_v0` is only ever available when linking libc++abi, so this program should be c++ (it should fail to link if its written in C I think).